src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -43,7 +43,7 @@ public boolean equals(Object obj) {
 
     public int hashCode() {
         int resual = 31;
-        resual = resual * (width * 3 * 4 / 2) + color.hashCode() * 2;
+        resual = resual * (width * 3 * 4 / 2) + length + color.hashCode() * 2;
```

Comment:
3 * 4 / 2 - what's the magic? it is always 12, isn't it? check each field for `null` and use their hashcodes to create this one